src/main/java/de/tum/in/www1/artemis/web/websocket/team/ParticipationTeamWebsocketService.java

```diff
@@ -197,11 +214,35 @@ else if (submission instanceof TextSubmission textSubmission && exercise instanc
             throw new IllegalArgumentException("Submission type '" + submission.getType() + "' not allowed.");
         }
 
+        if (syncTeammates) {
+            // update the last action date for the user and send out list of team members
+            updateValue(lastActionTracker, participationId, principal.getName());
+            sendOnlineTeamStudents(participationId);
+
+            SubmissionSyncPayload payload = new SubmissionSyncPayload(submission, user);
+            websocketMessagingService.sendMessage(getDestination(participationId, topicPath), payload);
+        }
+    }
+
+    private void patchSubmission(@DestinationVariable Long participationId, @Payload SubmissionPatch submissionPatch, Principal principal, String topicPath) {
+        // Without this, custom jpa repository methods don't work in websocket channel.
+        SecurityUtils.setAuthorizationObject();
+
+        final StudentParticipation participation = studentParticipationRepository.findByIdWithEagerTeamStudentsElseThrow(participationId);
```

Comment:
You load a lot of database objects here. This looks quite expensive. Is it really needed to distribute the patch? When it comes to websocket synchronization, we should keep it as lightweight as possible.

Reply:
This particular query is needed to ensure that the student issuing the patch is indeed a participant. I removed the rest of the database queries of this method though as they were not required.

Reply:
Looks better now. If you only need to check access rights, then it would make sense to create a custom query. Loading the data into Java could take some time and a query within the database is definitely faster. You could check if there is already an existing query and ask @MaximilianAnzinger for help if needed

Reply:
Since the performance benefit of such a query would be to not load any data but merely check existence of some record, I checked all such queries in the repository and couldn't find any, so I added a query for that purpose. It checks whether a participation with given ID exists such that student with given login is allowed to participate in it.